File: solutions/Python/334.IncreasingTripletSubsequence.py
class Solution:
    def increasingTriplet(self, nums: List[int]) -> bool:
        tripletI, tripletJ = math.inf, math.inf

        for num in nums:
            if num <= tripletI:
                tripletI = num
            elif num <= tripletJ:
                tripletJ = num
            else:
                return True

        return False

File: solutions/JavaScript/334.IncreasingTripletSubsequence.js
var increasingTriplet = function (nums) {
    let tripletI = Infinity
    let tripletJ = Infinity

    for (let num of nums) {
        if (num <= tripletI) {
            tripletI = num;
        } else if (num <= tripletJ) {
            tripletJ = num;
        } else {
            return true;
        }
    }

    return false;
};